flytekit/core/base_task.py

```diff
@@ -477,6 +478,8 @@ def __init__(
                 execution of the task. Supplied as a dictionary of key/value pairs
             disable_deck (bool): (deprecated) If true, this task will not output deck html file
             enable_deck (bool): If true, this task will output deck html file
+            additional_decks (Optional[List[str]]): List of additional decks besides [timeline and source code] to be
```

Comment:
Should we define it as a List[DeckFields] instead?

Reply:
I am +0 on using enums if we reimport it in `flytekit.deck`. Specifically:

```python
from flytekit.deck import PrebuiltDeck

@task(decks=[PrebuiltDeck.SourceCode, PrebuiltDeck.Inputs], enable_deck=True)
```

Enums is nice from a programming point of view, but they can make the feature harder to discover. For example, using strings means no additional imports:

```python
@task(decks=["source_code", "inputs"], enable_deck=True)
```

@novahow From a user point of view, do you have a preference?

Reply:
@thomasjpfan  Yeah I think using strings is fine.

Reply:
I prefer enum. how do users know what other decks we have? Using string is also error-prone. what if people use `sourceCode` or `Input`

Reply:
We list the strings out in the docstring and remember to add new ones to the list. If someone uses `sourceCode` or `Input`, then we raise an error.

For me, I come from the PyData world, where most APIS are strings. For example, SciPy's [`minimize)` function](https://docs.scipy.org/doc/scipy/reference/optimize.html#local-multivariate-optimization) uses string for `method`. The newly release [torch.comple](https://pytorch.org/docs/stable/generated/torch.compile.html#torch-compile) uses strings for `backend` and `mode`.

Personally, I prefer enums too, but I think most users in the Python data space are more familiar with strings.